src/polysolve/nonlinear/Solver.cpp

```diff
@@ -276,161 +273,137 @@ namespace polysolve::nonlinear
             objFunc.solution_changed(x);
         }
 
-        const auto g_norm_tol = this->m_stop.gradNorm;
-        this->m_stop.gradNorm = first_grad_norm_tol;
-
-        StopWatch stop_watch("nonlinear solver", this->total_time, m_logger);
+        StopWatch stop_watch("nonlinear solver", total_time, m_logger);
         stop_watch.start();
 
         m_logger.debug(
             "Starting {} with {} solve f₀={:g} "
             "(stopping criteria: max_iters={:d} Δf={:g} ‖∇f‖={:g} ‖Δx‖={:g})",
             descent_strategy_name(), m_line_search->name(),
-            objFunc.value(x), this->m_stop.iterations,
-            this->m_stop.fDelta, this->m_stop.gradNorm, this->m_stop.xDelta);
+            objFunc(x), m_stop.iterations,
+            m_stop.fDelta, m_stop.gradNorm, m_stop.xDelta);
 
-        update_solver_info(objFunc.value(x));
-        objFunc.post_step(PostStepData(this->m_current.iterations, solver_info, x, grad));
-
-        int f_delta_step_cnt = 0;
-        double f_delta = 0;
-
-        // Used for logging
-        double xDelta = 0, gradNorm = 0;
+        update_solver_info(objFunc(x));
+        objFunc.post_step(PostStepData(m_current.iterations, solver_info, x, grad));
 
         do
         {
             m_line_search->set_is_final_strategy(m_descent_strategy == m_strategies.size() - 1);
 
-            this->m_current.xDelta = NaN;
```

Comment:
Why removing those?

Reply:
Now, it always sets these variables, so there is no need to nan them.